core/frontend/src/components/nmea-injector/NMEASocketCreationDialog.vue

```diff
@@ -0,0 +1,155 @@
+            :items="available_socket_kinds"
+            label="Socket kind"
+            :rules="[validate_required_field]"
+            no-data-text="No serial ports available"
+            :loading="updating_serial_ports"
+          />
+
+          <v-text-field
+            v-model="nmea_socket.port"
+            :rules="[validate_required_field, is_socket_port]"
+            label="Socket port"
+          />
+
+          <v-text-field
+            v-model="nmea_socket.component_id"
+            :counter="3"
+            label="Mavlink component ID"
+            :rules="[validate_required_field, is_component_id]"
+          />
+
+          <v-btn
+            color="success"
+            class="mr-4"
+            @click="createNMEASocket"
+          >
+            Create
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+
+import NMEASocketStore from '@/store/nmea-injector'
+import NotificationStore from '@/store/notifications'
+import { SocketKind } from '@/types/common'
+import { nmea_injector_service } from '@/types/frontend_services'
+import { LiveNotification, NotificationLevel } from '@/types/notifications'
+import { VForm } from '@/types/vuetify'
+import {
+  isIntegerString, isNotEmpty, isSocketPort,
+} from '@/utils/pattern_validators'
+
+const notification_store: NotificationStore = getModule(NotificationStore)
+const nmea_injector_store: NMEASocketStore = getModule(NMEASocketStore)
+
+export default Vue.extend({
+  name: 'NMEASocketCreationDialog',
+  model: {
+    prop: 'show',
+    event: 'change',
+  },
+  props: {
+    show: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      nmea_socket: {
+        kind: SocketKind.UDP,
+        port: 27000 as (number | null),
+        component_id: 220,
+      },
+    }
+  },
+  computed: {
+    form(): VForm {
+      return this.$refs.form as VForm
+    },
+    available_socket_kinds(): {value: SocketKind, text: string}[] {
+      return Object.entries(SocketKind).map(
+        (kind) => ({ value: kind[1], text: kind[0] }),
+      )
+    },
+  },
+  methods: {
+    validate_required_field(input: string | number): (true | string) {
+      const string_input = String(input)
+      return isNotEmpty(string_input) ? true : 'Required field.'
+    },
+    is_socket_port(input: string): (true | string) {
+      if (!isIntegerString(input)) {
+        return 'Please use an integer value.'
+      }
+      const int_input = parseInt(input, 10)
+      return isSocketPort(int_input) ? true : 'Invalid port.'
+    },
+    is_component_id(input: string): (true | string) {
+      if (!isIntegerString(input)) {
+        return 'Please use an integer value.'
+      }
+      const int_input = parseInt(input, 10)
+      const is_in_id_range = int_input >= 0 && int_input <= 255
```

Comment:
It should be under 250 (the last component ID) and over 25 (first user component ID)
For more information: https://mavlink.io/en/messages/minimal.html#MAV_COMPONENT
It would be nice to add such comments as well